pvlib/iam.py

```diff
@@ -940,3 +942,383 @@ def schlick_diffuse(surface_tilt):
+    # appear in the physical model as a product, so it is enough to
+    # optimize for just L, and to fix K=4)
+    # set lower bound for n at 1.0 so that x-intercept will be at 90
+    # order for Powell's method depends on a_r value
+    bounds = [(1e-6, 0.08), (1.05, 2)]  # L, n
+    guess = [0.002, 1.1]  # L, n
+    # get better results if we reverse order to n, L at high a_r
+    if a_r > 0.22:
+        bounds.reverse()
+        guess.reverse()
+
+    # the product of K and L is more important in determining an initial
+    # guess for the location of the minimum, so we pass L in first
+    def residual_function(target_params):
+        # unpack target_params for either search order
+        if target_params[0] < target_params[1]:
+            # L will always be less than n
+            L, n = target_params
+        else:
+            n, L = target_params
+        return _residual(aoi, martin_ruiz_iam, physical, [n, 4, L], weight)
+
+    return residual_function, guess, bounds
+
+
+def _minimize(residual_function, guess, bounds, xtol):
+    if xtol is not None:
+        options = {'xtol': xtol}
+    else:
+        options = None
+    with np.errstate(invalid='ignore'):
+        optimize_result = minimize(residual_function, guess, method="powell",
+                                   bounds=bounds, options=options)
+
+    if not optimize_result.success:
+        try:
+            message = "Optimizer exited unsuccessfully:" \
+                      + optimize_result.message
+        except AttributeError:
+            message = "Optimizer exited unsuccessfully: \
+                       No message explaining the failure was returned. \
+                       If you would like to see this message, please \
+                       update your scipy version (try version 1.8.0 \
+                       or beyond)."
+        raise RuntimeError(message)
+
+    return optimize_result
+
+
+def _process_return(target_name, optimize_result):
+    if target_name == "ashrae":
+        target_params = {'b': optimize_result.x.item()}
+
+    elif target_name == "martin_ruiz":
+        target_params = {'a_r': optimize_result.x.item()}
+
+    elif target_name == "physical":
+        L, n = optimize_result.x
+        # have to unpack order because search order may be different
+        if L > n:
+            L, n = n, L
+        target_params = {'n': n, 'K': 4, 'L': L}
+
+    return target_params
+
+
+def _min_scipy():
+    '''convert and fit require scipy>=1.5.0
+    '''
+    from scipy import __version__
+    major, minor = __version__.split('.')[:2]
+    if (int(major) >= 1) & (int(minor) >= 5):
+        return True
+    else:
+        warnings.warn('iam.convert and iam.fit require scipy>=1.5.0')
+        return False
+
+
+def convert(source_name, source_params, target_name, weight=_sin_weight,
+            fix_n=True, xtol=None):
+    """
+    Convert a source IAM model to a target IAM model.
```

Comment:
```suggestion
    Convert source IAM model parameters to target IAM model parameters.
```

Reply:
I'll quibble with this edit, we're not converting `b` to `a_r`, we're estimating `a_r` from the source model's output.

Reply:
Don't worry about it.